### PR TITLE
Utilities and fixes for HipPy alignment

### PR DESCRIPTION
**Pull request by hroskes**

### CODE REVIEW

**Comment by fabiocos** [comment]
please test

**Review by pohsun — CHANGES_REQUESTED** [review]

**Comment by pohsun** [comment]
please test

**Comment by pohsun** [comment]
+1

File: Alignment/HIPAlignmentAlgorithm/scripts/hippyaddtobaddatafiles.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      with OneAtATime(badfilelist+".tmp", 2) as f:
+        with open(badfilelist) as f:
+          contents = set(f.read().split())
+        if filename in contents:
+          raise RuntimeError(filename+"\nis already in\n"+badfilelist+"\n\nExiting to avoid an infinite loop.  Maybe you have this running on the same cfg file multiple times?")
+        contents.add(filename)
+        contents = sorted(contents)
+        with open(badfilelist, "w") as f:
+          f.write("\n".join(contents)+"\n")
+        print "found and added a bad file:\n"+filename
```
Comment (fabiocos) [inline Alignment/HIPAlignmentAlgorithm/scripts/hippyaddtobaddatafiles.py]: @hroskes could you please make this python3 compliant adding parentheses to the print statements and 
from __future__ import print_function
at the beginning?

**Comment by pohsun** [comment]
please test

**Comment by pohsun** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**